qiskit_qec/decoders/hdrg_decoders.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
```

Comment:
If there are going to be multiple hdrg decoders, do we want to just make a subfolder
`decoders/hdrg_decoders/clustering_decoder.py`  so each decoder still gets their own file?